src/vs/editor/common/viewLayout/viewLinesViewportData.ts

```diff
@@ -47,6 +47,7 @@ export class ViewportData {
 	private readonly _model: IViewModel;
 
 	public readonly lineHeight: number;
+	public readonly specialLineHeights: Map<number, number>;
```

Comment:
Please don't add this field. Do not spread out the logic of the special line heights. Let's use a single way to get to it via `this._context.viewLayout.getLineHeightForLineNumber`